agent/agent.go

```diff
@@ -1880,6 +1880,12 @@ func (a *Agent) AddService(service *structs.NodeService, chkTypes []*structs.Che
 		}
 	}
 
+	// Set default weights if not specified. This is important as it ensures AE
+	// doesn't consider the service different since it has nil weights.
```

Comment:
Is this because on the server side they end up getting defaulted? This seems fine to me regardless, just trying to understand why there is a problem.

Reply:
Yep I think so as I understand it.